techmatters/hrm
# CHI-1592: Ensure retired resources are not returned is searches, remove sequence columns

## Description

PR to clean a few things up

Ensure deleted resources are not returned from the search index or direct DB queries by:
* Adding a 'deletedAt' column to resources
* Ensuring resources are not returned from UI facing endpoints if the deletedAt value is set
* Mapping the 'retired' value in KHP resources to the deletedAt field
* Ensuring that any resources sent for indexing with a deletedAt field are actually removed from the search index rather than indexed

Remove the updateSequenceId columns  and their associated triggers / functions from the DB schema

Remove the placeholder search (searchByName)

Let me know if you'd prefer thgis splitting up, it shouldn't be too hard to do

### Checklist


### Related Issues
CHI-1592, CHI-2005

### Verification steps

Automated tests should verify the basic functions
Indexing functionality can be tested by:
* Manually adding a deletedAt date to a resource in the DB
* Reindex that resource using the admin script
* Search for it and ensure it is not part of the results
* Removed the deletedAt date
* Reindex the same resource again
* Ensure it is back in the results
